pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -112,13 +112,18 @@ type Adapter struct {
 // Once the component has started, it will sync the source code to it.
 func (a Adapter) Push(parameters common.PushParameters) (err error) {
 
-	a.deployment, err = a.Client.GetOneDeployment(a.ComponentName, a.AppName)
+	// Get the Dev deployment:
+	// Since `odo deploy` can theoretically deploy a deployment as well with the same instance name
+	// we make sure that we are retrieving the deployment with the Dev mode, NOT Deploy.
+	selectorLabels := componentlabels.GetLabels(a.ComponentName, a.AppName, false)
+	selectorLabels[componentlabels.OdoModeLabel] = componentlabels.ComponentDevName
```

Comment:
wouldn't it be better to include `OdoModeLabel` in `componentlabels.GetLabels`?
It will be safer and it it will be automatically used everywhere where it is needed.

Reply:
I don't think so unless there is a refactor with the GetLabels and how we do it elsewhere (it'd break `odo push` tests).

I did think of your solution, but found I was changing WAYYY too many places where the GetLabels code is. Especially in parts of the code where you would have to modify multiple functions just to pass in "Dev" or "Deploy" from the `pkg/cli` layer.